pkg/scheduler/testing/framework/fake_extender.go

```diff
@@ -191,10 +192,11 @@ func (f *FakeExtender) ProcessPreemption(
 		nodeNameToVictimsCopy[k] = v
 	}
 
+	logger := klog.Background()
```

Comment:
If I fetched logger up again, I would need to modify the `ProcessPreemption` function, which would result in a change to the `Extender` interface（pkg/scheduler/framework/extender.go:27）, which looks at annotations for use by external processes(This interface hasn't changed in three years), which I am afraid would cause an interruption to the external process

// pkg/scheduler/extender.go:42
The actual implementation struct I found in the code is HTTPExtender.

Please check with the reviewer to see if there will be any impact on external processes and if I don't need to modify the extender interface.

Moreover, this change is only used by the fake code, and there is not much need to change the interface.

Reply:
`HTTPExtender.ProcessPreemption` does a `POST`, i.e. it does network communication. I think it would be beneficial to add a context to the `Extender` API, which (at some point) could be used to communicate timeouts.

Just my 2 cents, I'll defer to the scheduler developers here.

For now, better use `klog.TODO()` with a comment about "if Extender.ProcessPreemption ever gets extended with a context parameter, then the logger should be retrieved from that".

Reply:
Please don't modify the `Extender` interface, we will not maintainer the extender in the long term, and we also hope not breaking existing users. +1 with pohly.